File: go-odds/internal/evaluator/street.go
package evaluator

import "fmt"

// Street represents a betting round in a Texas Hold'em hand.
type Street string

const (
	Preflop  Street = "PREFLOP"
	Flop     Street = "FLOP"
	Turn     Street = "TURN"
	River    Street = "RIVER"
	Showdown Street = "SHOWDOWN"
)

// HasBoardCards returns true if community cards have been dealt.
func (s Street) HasBoardCards() bool {
	return s == Flop || s == Turn || s == River || s == Showdown
}

// IsEquityStreet returns true if equity calculation is meaningful.
func (s Street) IsEquityStreet() bool {
	return s == Flop || s == Turn || s == River
}

// Next returns the following street, or an error if already at Showdown.
func (s Street) Next() (Street, error) {
	switch s {
	case Preflop:
		return Flop, nil
	case Flop:
		return Turn, nil
	case Turn:
		return River, nil
	case River:
		return Showdown, nil
	default:
		return "", fmt.Errorf("no street after %q", s)
	}
}

// ExpectedBoardCards returns how many community cards should be present.
func (s Street) ExpectedBoardCards() int {
	switch s {
	case Flop:
		return 3
	case Turn:
		return 4
	case River, Showdown:
		return 5
	default:
		return 0
	}
}

File: java-api/src/main/java/com/poker/domain/model/Street.java
package com.poker.domain.model;

/**
 * The betting streets of a Texas Hold'em hand, in order.
 *
 * <p>SHOWDOWN is a terminal pseudo-street — it is reached after RIVER betting
 * closes and indicates that players must reveal their hole cards.
 */
public enum Street {
    PREFLOP,
    FLOP,
    TURN,
    RIVER,
    SHOWDOWN;

    /** True if community cards have been dealt (flop or later). */
    public boolean hasBoardCards() {
        return this == FLOP || this == TURN || this == RIVER || this == SHOWDOWN;
    }

    /** True if the equity calculation is meaningful (flop/turn/river). */
    public boolean isEquityStreet() {
        return this == FLOP || this == TURN || this == RIVER;
    }

    /** Returns the next street, or throws if already at SHOWDOWN. */
    public Street next() {
        Street[] values = values();
        if (this.ordinal() >= values.length - 1) {
            throw new IllegalStateException("No street after " + this);
        }
        return values[this.ordinal() + 1];
    }
}
